pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -28,8 +37,41 @@ type DatasourceInfo struct {
 	SecretKey string
 }
 
-func NewCloudWatchExecutor(dsInfo *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
-	return &CloudWatchExecutor{}, nil
+func (e *CloudWatchExecutor) getLogsClient(region string) (*cloudwatchlogs.CloudWatchLogs, error) {
+	e.mux.Lock()
+	defer e.mux.Unlock()
+
+	if logsClient, ok := e.logsClientsByRegion[region]; ok {
+		return logsClient, nil
+	}
+
+	dsInfo := retrieveDsInfo(e.DataSource, region)
+	newLogsClient, err := retrieveLogsClient(dsInfo)
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
- [ ] As @bergquist pointed out for another block, it's better to co-locate the error check with the originating statement, since they logically belong together:

```suggestion
	newLogsClient, err := retrieveLogsClient(dsInfo)
	if err != nil {
		return nil, err
	}
```

Reply:
Why resolve this comment? I think it's a fair concern. Might seem like a nitpick, but it makes a difference when reviewing. It does happen that there are stale error checks around, that are left behind after merges and should be deleted.